linalg/solvers.cpp

```diff
@@ -1319,6 +1319,54 @@ void NewtonSolver::Mult(const Vector &b, Vector &x) const
    final_norm = norm;
 }
 
+void NewtonSolver::CheckJacobian(Vector &x, Array<int> &ess_tdof)
```

Comment:
Why is this function in NewtonSolver and not in Operator?

Reply:
I discussed the same idea with @v-dobrev today and we concluded this method fits better in the Operator itself.

This also enables to implement the action of the gradient using finite differences for tests in the solver method.